dnsimple/templates.go

```diff
@@ -127,3 +127,107 @@ func (s *TemplatesService) DeleteTemplate(accountID string, templateID string) (
 	templateResponse.HttpResponse = resp
 	return templateResponse, nil
 }
+
+// Template Records
+
+// TemplateRecord represents a DNS record for a template in DNSimple.
+type TemplateRecord struct {
+	ID         int    `json:"id,omitempty"`
+	TemplateID int    `json:"template_id,omitempty"`
+	Name       string `json:"name"`
+	Content    string `json:"content,omitempty"`
+	TTL        int    `json:"ttl,omitempty"`
+	Type       string `json:"type,omitempty"`
+	Priority   int    `json:"priority,omitempty"`
+	CreatedAt  string `json:"created_at,omitempty"`
+	UpdatedAt  string `json:"updated_at,omitempty"`
+}
+
+// TemplateRecordResponse represents a response from an API method that returns a TemplateRecord struct.
+type TemplateRecordResponse struct {
+	Response
+	Data *TemplateRecord `json:"data"`
+}
+
+// TemplateRecordsResponse represents a response from an API method that returns a collection of TemplateRecord struct.
+type TemplateRecordsResponse struct {
+	Response
+	Data []TemplateRecord `json:"data"`
+}
+
+func templateRecordPath(accountID string, templateID string, templateRecordID string) string {
+	if templateRecordID != "" {
+		return fmt.Sprintf(templatePath(accountID, templateID)+"/records/%v", templateRecordID)
```

Comment:
If you use `Sprintf`, then the is no need to concat. I'd write 

```go
return fmt.Sprintf("%v/records/%v", templatePath(accountID, templateID), templateRecordID)
```

You can probably also merge the two `templatePath` calls into something like

```go
func templateRecordPath(accountID string, templateID string, templateRecordID string) string {
    path := fmt.Sprintf("%v/records", templatePath(accountID, templateID))
    if templateRecordID != "" {
        path += fmt.Sprintf("/%s", templateRecordID)
    }
    return path
}